model/service/StructuredVariablesToItemStateService.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ */
+
+declare(strict_types=1);
+
+namespace oat\ltiOutcomeUi\model\service;
+
+use oat\oatbox\service\ConfigurableService;
+use oat\taoOutcomeUi\helper\ResponseVariableFormatter;
+use taoResultServer_models_classes_ResponseVariable as ResponseVariable;
+
+class StructuredVariablesToItemStateService extends ConfigurableService
+{
+    public const SERVICE_ID = 'ltiOutcomeUi/StructuredVariablesToItemStateService';
+
+    private const RESULT_KEY_URI = 'uri';
+    private const RESULT_KEY_ATTEMPT = 'attempt';
+    private const RESULT_KEY_RESPONSE_VARIABLE = 'taoResultServer_models_classes_ResponseVariable';
+
+    public function format(array $testResultVariables): array
+    {
+        $formatted = [];
+        foreach ($testResultVariables as $resultVariables) {
+            $itemResults = $this->formatResultVariables($resultVariables);
+            if ($itemResults) {
+                $itemUri = $resultVariables[self::RESULT_KEY_URI];
+                $formatted[$itemUri][$resultVariables[self::RESULT_KEY_ATTEMPT]] = $itemResults;
+            }
+        }
+
+        return $formatted;
+    }
+
+    private function formatResultVariables(array $resultVariables): ?array
+    {
+        if (false === isset($resultVariables[self::RESULT_KEY_URI], $resultVariables[self::RESULT_KEY_ATTEMPT])) {
```

Comment:
Is this unit tested?

Reply:
I will add this case to the test

Reply:
`isset()` always returns a boolean, so strict checking (`===`) is not needed here: Checking the return value from `isset` directly may be a bit clearer.

```suggestion
        if (!isset($resultVariables[self::RESULT_KEY_URI], $resultVariables[self::RESULT_KEY_ATTEMPT])) {
```

Reply:
We must always use the stripe check because in PHP we always have problems with the types, many times we write the code as Javascript and that does not make the code clear, with the negation we only simplify the expression but the code is not clear, the `false === X` is read much better than `!X`, we must try to write the code so that it reads better.

Reply:
I don't agree with this. More readable (cause shorter) is expression `!x`. I support @hectoras comment. I don't understand argument about types and js code style.

Reply:
Well, I don't agree with using the exclamation mark either, how do we reach the conclusion?

Reply:
It's a boolean condition. The `if` in it's nature compares boolean condition, it's more natural to use exclamation mark. It's a waste of time to continue discussion. Let's use exclamation mark, period.

https://stackoverflow.com/questions/3598043/best-practice-iffoo-false-or-iffoo